apps/platform-app/src/backend/linkedin-details/linkedin-details.service.ts

```diff
@@ -0,0 +1,49 @@
+import { prisma } from '@/lib/prismaClient'
+
+export const findLinkedinUsernamesByUsernameBase = async (
+  baseUsername: string,
+): Promise<string[]> => {
+  const usernames = await prisma.linkedInDetails.findMany({
+    where: {
+      username: {
+        startsWith: baseUsername,
+      },
+    },
+    select: {
+      username: true,
+    },
+  })
+
+  return usernames.map(
+    (usernameObject: { username: string }) => usernameObject.username,
+  )
+}
+
+export async function createLinkedInDetailsForUser(
+  userId: string,
+  username: string,
+) {
+  // Check if the user exists and does not have LinkedInDetails
+  const user = await prisma.user.findUnique({
+    where: { id: userId },
+    include: { linkedinDetails: true },
+  })
+
+  if (!user) {
+    throw new Error('User not found')
```

Comment:
**issue**: Consider providing a more descriptive error message when the user is not found to aid in debugging.